server/src/main/java/org/elasticsearch/plugins/scanners/NameToPluginInfo.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.plugins.scanners;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+public class NameToPluginInfo {
+
+    // plugin name to NamedPluginInfo
+    private Map<String, PluginInfo> namedPluginInfoMap = new HashMap<>();
```

Comment:
field should be `final`.
Is this a good class for a `record`, that will handle toString/equals/hashcode for us?

Reply:
I think a record here is a good idea